kaggle_environments/envs/chess/chess.js

```diff
@@ -21,6 +43,76 @@ async function renderer(context) {
     parent.appendChild(canvas);
   }
 
+  // Create the Download PGN button
+  let downloadButton = parent.querySelector("#copy-pgn");
+  if (!downloadButton) {
+    try {
+      downloadButton = document.createElement("button");
+      downloadButton.id = "copy-pgn";
+      downloadButton.textContent = "Copy PGN";
+      downloadButton.style.position = "absolute";
+      downloadButton.style.top = "10px";
+      downloadButton.style.left = "10px";
+      downloadButton.style.zIndex = 1;
+      parent.appendChild(downloadButton);
+
+      const board = environment.steps[step][0].observation.board;
+      const info = environment.info;
+      const agent1 = info?.TeamNames?.[0] || "Agent 1";
+      const agent2 = info?.TeamNames?.[1] || "Agent 2";
+      const game = new Chess(board);
+      let result = environment.rewards;
+      if (result.some(r => r === undefined || r === null)) {
+        result = result.map(r => r === undefined || r === null ? 0 : 1)
+      }
+
+      game.header(
+        "Event",
+        "FIDE & Google Efficient Chess AI Challenge (https://www.kaggle.com/competitions/fide-google-efficiency-chess-ai-challenge)",
+        "White",
+        agent1,
+        "Black",
+        agent2,
+        "Result",
+        result.join(" - ")
+      );
+
+      const openingIdx = OPENINGS.indexOf(board);
+      const moves = MOVES[openingIdx].split(" ");
+
+      for (let i = 0; i < moves.length; i++) {
+        const move = moves[i];
+        game.move({ from: move.slice(0, 2), to: move.slice(2, 4) });
+      }
+
+      for (let i = 1; i < environment.steps.length; i++) {
```

Comment:
double checking this should be 1-indexed

Reply:
yes, the first "step" of the game is showing the board state to both agents